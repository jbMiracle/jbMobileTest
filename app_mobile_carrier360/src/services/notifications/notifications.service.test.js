import {
  presentLocalNotification,
  presentLocalNotificationWithActions,
  scheduleLocalNotification,
  scheduleLocalNotificationWithActions,
  ActionCategoryIds,
} from '.';

const basicOptions = {
  title: 'Test Title',
  body: 'test body',
};

const basicScheduledOptions = {
  ...basicOptions,
  id: 'test-notification-id',
  fire_date: 1577836800,
};

describe('presentLocalNotification()', () => {
  it('should pass the appropriate options to FCM.presentLocalNotification()', () => {
    const combinedOptions = presentLocalNotification(basicOptions);

    expect(combinedOptions).toMatchSnapshot();
  });
});

describe('presentLocalNotificationWithActions()', () => {
  it('should pass the appropriate options to FCM.presentLocalNotification()', () => {
    const combinedOptions = presentLocalNotificationWithActions(ActionCategoryIds.exampleA, basicOptions);

    expect(combinedOptions).toMatchSnapshot();
  });

  it('should throw an error when passed an unregistered action category ID', () => {
    expect(() => {
      presentLocalNotificationWithActions('badActionId', basicOptions);
    }).toThrow();
  });
});

describe('scheduleLocalNotification()', () => {
  it('should pass the appropriate options to FCM.scheduleLocalNotification()', () => {
    const combinedOptions = scheduleLocalNotification(basicScheduledOptions);

    expect(combinedOptions).toMatchSnapshot();
  });
});

describe('scheduleLocalNotificationWithActions()', () => {
  it('should pass the appropriate options to FCM.scheduleLocalNotification()', () => {
    const combinedOptions = scheduleLocalNotificationWithActions(ActionCategoryIds.exampleA, basicScheduledOptions);

    expect(combinedOptions).toMatchSnapshot();
  });
});
