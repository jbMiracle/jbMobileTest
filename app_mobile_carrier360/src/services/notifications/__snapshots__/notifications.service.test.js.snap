// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presentLocalNotification() should pass the appropriate options to FCM.presentLocalNotification() 1`] = `
Object {
  "body": "test body",
  "color": "#007DBA",
  "priority": "high",
  "show_in_foreground": true,
  "title": "Test Title",
}
`;

exports[`presentLocalNotificationWithActions() should pass the appropriate options to FCM.presentLocalNotification() 1`] = `
Object {
  "body": "test body",
  "click_action": "com.jbhunt.carrier360.exampleA",
  "color": "#007DBA",
  "priority": "high",
  "show_in_foreground": true,
  "title": "Test Title",
}
`;

exports[`scheduleLocalNotification() should pass the appropriate options to FCM.scheduleLocalNotification() 1`] = `
Object {
  "body": "test body",
  "color": "#007DBA",
  "fire_date": 1577836800,
  "id": "test-notification-id",
  "priority": "high",
  "show_in_foreground": true,
  "title": "Test Title",
}
`;

exports[`scheduleLocalNotificationWithActions() should pass the appropriate options to FCM.scheduleLocalNotification() 1`] = `
Object {
  "body": "test body",
  "click_action": "com.jbhunt.carrier360.exampleA",
  "color": "#007DBA",
  "fire_date": 1577836800,
  "id": "test-notification-id",
  "priority": "high",
  "show_in_foreground": true,
  "title": "Test Title",
}
`;
